packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -100,27 +109,48 @@ protected override ConstructorProvider[] BuildConstructors()
             return [constructor];
         }
 
-        private IReadOnlyList<ParameterProvider> BuildConstructorParameters()
+        private (IReadOnlyList<ParameterProvider> Parameters, ConstructorInitializer? Initializer) BuildConstructorParameters()
```

Comment:
Lets open a TODO / issue link on how to reuse this code with MrwSerializationProvider

Reply:
https://github.com/microsoft/typespec/issues/3871